### ISSUE TITLE
IsEventSourceLoggingEnabled Missing on Linux ARM

### ISSUE THREAD

**Issue by manu-silicon** [dotnet/coreclr#3857]
I'm having an issue on Linux Arm where it cannot find an ID for method ```IsEventSourceLoggingEnabled``` at line 542 of ecall.cpp (https://github.com/dotnet/coreclr/blob/master/src/vm/ecall.cpp#L542)

Turns out that FEATURE_EVENT_TRACE is disabled for Linux ARM by this PR #2935. Should this be reverted back to allow FEATURE_EVENT_TRACE for all platforms? Or should we change mscorlib to not call it when it is not defined?

**Comment by manu-silicon** [comment]
For the time being, I've enabled FEATURE_EVENT_TRACE without visible ill effect.

**Comment by manu-silicon** [comment]
@jkotas Any idea about this one?

**Comment by jkotas** [comment]
There is a mismatch when the FEATURE_EVENT_TRACE is defined on managed vs. managed sides.

I think it may be best to:
- Enable XplatEventLogger on managed side for all non-Windows platforms (ie change the ifdef around it to `FEATURE_PAL`). XplatEventLogger is very small, and so it is not a problem to have compiled in even when it is not strictly necessary.
- Add dummy unmanaged implementation of the two internal calls needed by XplatEventLogger (XplatEventSourceLogger::IsEventSourceLoggingEnabled and XplatEventSourceLogger::LogEventSource) for !FEATURE_EVENT_TRACE.

**Comment by jkotas** [comment]
This should make it easy to control whether the x-plat event tracing is enabled through the one central place in top-level CMakeLists.txt.

**Comment by manu-silicon** [comment]
@jkotas Will do that. Any special place to add the dummy unmanaged implementation?

**Comment by jkotas** [comment]
We usually add those right next to the real ones. Or in this case - it may be best to just ifdef out the bodies of the real ones, like:

```
BOOL QCALLTYPE XplatEventSourceLogger::IsEventSourceLoggingEnabled()
{
    QCALL_CONTRACT;

    BOOL retVal = FALSE;

#if defined(FEATURE_EVENTSOURCE_XPLAT)
    BEGIN_QCALL;
    retVal = XplatEventLogger::IsEventLoggingEnabled();
    END_QCALL;
#endif
    
    return retVal;
}
```

**Comment by manu-silicon** [comment]
Done so in PR #3879.

**Comment by manu-silicon** [comment]
Closed via #3879
